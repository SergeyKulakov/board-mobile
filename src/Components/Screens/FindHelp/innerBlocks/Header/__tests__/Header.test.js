import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import { Header as Component } from '../index'

describe('FindHelp.EmptyScreen', () => {
  const defaultProps = {
    text: {
      title: 'title',
      getTabTitle: key => key,
    },
    activeScreenId: 0,
    onClickSearch: () => null,
    onClickBack: () => null,
    onClickHamburger: () => null,
    onClickFilter: () => null,
    onClickNavItem: () => null,
    onClickSort: () => null,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('select tab', () => {
    const props = {
      ...defaultProps,
      onClickNavItem: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    const { length } = wrapper.find('TabNavigationElement')

    for (let i = 0; i < length; i++) {
      wrapper
        .find('TabNavigationElement')
        .get(i)
        .props.onClick()
    }

    expect(props.onClickNavItem).toHaveBeenCalledTimes(2)
  })
})
