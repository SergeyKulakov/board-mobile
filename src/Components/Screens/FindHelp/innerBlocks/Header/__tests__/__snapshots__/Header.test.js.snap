// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindHelp.EmptyScreen render snapshot 1`] = `
<Component>
  <Styled(View)>
    <Styled(View)>
      <Styled(View)
        mr={10}
      >
        <BackIcon
          onClick={[Function]}
        />
      </Styled(View)>
      <Styled(View)
        flex={2}
      >
        <Component
          onClick={[Function]}
        />
      </Styled(View)>
      <Styled(View)
        flex={10}
      >
        <ScreenTitle
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          title
        </ScreenTitle>
      </Styled(View)>
      <Styled(View)>
        <Icon
          color="#fff"
          name="search"
          onClick={[Function]}
        />
        <Styled(View)>
          <Icon
            color="#fff"
            name="filter"
            onClick={[Function]}
          />
        </Styled(View)>
        <Styled(Icon)
          onClick={[Function]}
        />
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <TabNavigationElement
        isActive={true}
        key="0"
        onClick={[Function]}
        title="list"
      />
      <TabNavigationElement
        isActive={false}
        key="1"
        onClick={[Function]}
        title="map"
      />
    </Styled(View)>
  </Styled(View)>
</Component>
`;
